Resources/public/Admin.js

```diff
@@ -69,6 +71,45 @@ var Admin = {
             overflow: 'scroll'
         });
     },
+    
+    setup_select: function(subject) {
+       jQuery('select', subject).each(function() {
+            var select = jQuery(this);
+            var fieldContainer = select.closest('div[class="field-container"]');
+            var editBtn = fieldContainer.find('a[class~="sonata_model_edit_btn"]');
+
+            if (editBtn.length) {
+                editBtn.rewriteBtn = function(choice) {
+                    if (choice) {
+                        var href = jQuery(choice).attr('data-sonata-admin-edit-href');
+                        if (href !== undefined && href !== '') {
+                            this.removeClass('disabled');
+                            this.attr('href', href);
+                            return;
+                        }
+                    }
+                    // disable edit link if choice is invalid or removed
+                    this.addClass('disabled');
+                    this.attr('href', null);
+                };
+                
+                // if an entity is already selected
+                var choice = select.find('option:selected:last');
```

Comment:
Also, ask yourself if you really must provide an edit button when multiple choices are allowed.
I would say no, since the user won't be able to choose which entry to edit (unless you keep track of the most recently selected choice for every input).